File: example01/command/ProvisionCommand.java
package boinsoft.example01.command;

import boinsoft.example01.*;
import org.osgi.service.component.annotations.*;

@Component(
    property = {"osgi.command.scope=provision", "osgi.command.function=provision"},
    immediate = true,
    service = Object.class)
public class ProvisionCommand {

  @Reference private ProvisionService provisionService_;

  public Handle start() {
    return provisionService_.start(new MachineSpecification(256));
  }
}

File: example02/command/ProvisionCommand.kt
package boinsoft.example02.command;

import boinsoft.example02.*;
import org.osgi.service.component.annotations.*;


@Component(
    property = ["osgi.command.scope=example02", "osgi.command.function=provision"],
    immediate = true,
    service = [Object::class])
class ProvisionCommand {
  
  @Reference
  var provisionService_: ProvisionService? = null

  fun start(): Handle? {
    return provisionService_?.start(MachineSpecification(256))
  }
}
